app/src/main/java/com/codingwithmitch/coroutineexamples/MainActivity.kt

```diff
@@ -4,77 +4,52 @@ import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
 import kotlinx.android.synthetic.main.activity_main.*
 import kotlinx.coroutines.*
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.Dispatchers.Main
 
 class MainActivity : AppCompatActivity() {
 
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
         button.setOnClickListener {
-            setTextOnMainThread("Click!")
+            // Ideally you should use launch anything here at all
+            GlobalScope.launch { setTextOnMainThread("Click!") }
             fakeApiRequest()
         }
     }
 
-    fun setTextOnMainThread(input: String){
-        CoroutineScope(Dispatchers.Main)
-            .launch { withContext(Dispatchers.Main){
-                val newText = text.text.toString() + "\n${input}"
-                text.setText(newText)
-            } }
+    private suspend fun setTextOnMainThread(input: String) {
+        withContext(Main) {
```

Comment:
You should also avoid launching new coroutines, and rather just switch contexts which can be done in any suspend function or suspend lambda